frontend/src/_components/CreateApp.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect, useContext } from 'react';
+import { toast } from 'react-hot-toast';
+import Modal from '../HomePage/Modal';
+import { ButtonSolid } from '@/_ui/AppButton/AppButton';
+import { appService } from '@/_services';
+import _ from 'lodash';
+import { validateAppName } from '@/_helpers/utils';
+
+export function CreateApp({ closeModal, createApp, show }) {
+  const [appName, setappName] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleCreateApp = async () => {
+    if (!errorText) {
+      setIsLoading(true);
+      try {
+        const success = await createApp(appName);
+        if (success === false) {
+          setErrorText('App name already exists');
+        } else {
+          setErrorText('');
+          closeModal();
+        }
+      } catch (error) {
+        toast.error(error);
+      }
+    }
+    setIsLoading(false);
+  };
+
+  const handleInputChange = (e) => {
+    const newAppName = e.target.value.trim();
+    const error = validateAppName(newAppName, 'App Name', true, false);
```

Comment:
I see `validateAppName` only takes two arguments but you have passed four and even second argument isn't expected data type.

Reply:
Thanks for your review @Ckbhatia , this is updated here. https://github.com/ToolJet/ToolJet/pull/7264